sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ExtRefService.java

```diff
@@ -112,41 +112,50 @@ public List<SclReportItem> updateAllExtRefIedNames(SCL scd) {
     }
 
     @Override
-    public List<SclReportItem> createDataSetAndControlBlocks(SCL scd) {
+    public List<SclReportItem> createDataSetAndControlBlocks(SCL scd, Set<FcdaCsvHelper.FcdaForDataSets> allowedFcdas) {
+        checkFcdaInitDataPresence(allowedFcdas);
         SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
         Stream<LDeviceAdapter> lDeviceAdapters = sclRootAdapter.streamIEDAdapters().flatMap(IEDAdapter::streamLDeviceAdapters);
-        return createDataSetAndControlBlocks(lDeviceAdapters);
+        return createDataSetAndControlBlocks(lDeviceAdapters, allowedFcdas);
     }
 
     @Override
-    public List<SclReportItem> createDataSetAndControlBlocks(SCL scd, String targetIedName) {
+    public List<SclReportItem> createDataSetAndControlBlocks(SCL scd, String targetIedName, Set<FcdaCsvHelper.FcdaForDataSets> allowedFcdas) {
```

Comment:
this method is not used isn't it ?

Reply:
As comment above, it is proposed for the community